cmake/Modules/CodeCoverage.cmake

```diff
@@ -170,7 +170,7 @@ function(ADD_CODE_COVERAGE)
 
     # Create Python coverage report
     add_custom_target(${Coverage_NAME}_py
-        COMMAND cp ${PROJECT_BINARY_DIR}/.coverage ${COVERAGE_DIR}/.coverage.nosetests || echo "WARNING: No nosetest coverage!"
```

Comment:
ok, so this actually changes the functionality - we *need* to rename the .coverage file generated by catkin nose tests into .coverage.something or it gets overwritten and lost during the combine step below. I'd suggest we keep this line AND add your line to copy other files over. Even though the .coverage file will get copied twice, the second copy will get overwritten during the combine.

Reply:
Thanks for the quick reply. Sorry I oversaw this issue. I'll recheck with a repo containing both pytest and nosetest just to make sure and get back at you!

Reply:
Yeah, no issue - I didn't find it until I tested against our private repo - and then I had to think about why it wasn't working before I remembered why I was renamed .coverage to .coverage.nosetest (there should probably be a comment added there too explaining the rename, if you don't end up adding it in this PR, I'll add it in a future one).

Reply:
Maybe we can remedy all concerns by writing something like this:
```
if (EXISTS "${PROJECT_BINARY_DIR}/.coverage")
   # Rename .coverage file generated by nosetests to avoid overwriting
   COMMAND mv ${PROJECT_BINARY_DIR}/.coverage ${PROJECT_BINARY_DIR}/.coverage.nosetests
endif()
COMMAND cp ${PROJECT_BINARY_DIR}/.coverage* ${COVERAGE_DIR}/
```

Reply:
Maybe we can remedy all concerns by writing something like this:
```
if (EXISTS "${PROJECT_BINARY_DIR}/.coverage")
   # Rename .coverage file generated by nosetests to avoid overwriting
   COMMAND mv ${PROJECT_BINARY_DIR}/.coverage ${PROJECT_BINARY_DIR}/.coverage.nosetests
endif()
COMMAND cp ${PROJECT_BINARY_DIR}/.coverage* ${COVERAGE_DIR}/
```

Reply:
@hslusarek that should work. Would suggest just adding "during combine step" to end of comment (so it is clear who/what would be overwriting).